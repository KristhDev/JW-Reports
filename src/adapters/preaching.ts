import { Preaching, PreachingEndpoint } from '../interfaces';

/**
 * Creates a new Preaching object from a PreachingEndpoint object.
 *
 * @param {PreachingEndpoint} preaching - The PreachingEndpoint object to convert.
 * @return {Preaching} The converted Preaching object.
 */
export const preachingAdapter = (preaching: PreachingEndpoint): Preaching => ({
    id: preaching.id,
    userId: preaching.user_id,
    day: preaching.day,
    initHour: preaching.init_hour,
    finalHour: preaching.final_hour,
    createdAt: preaching.created_at,
    updatedAt: preaching.updated_at
});