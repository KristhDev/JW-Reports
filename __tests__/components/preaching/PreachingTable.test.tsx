import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';

/* Components */
import { PreachingTable } from '../../../src/components/preaching';

/* Hooks */
import { usePreaching, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Setup */
import { navigateMock } from '../../../jest.setup';

/* Mocks */
import { preachingsStateMock, setSelectedPreachingMock } from '../../mocks';

/* Mock hooks */
jest.mock('../../../src/hooks/usePreaching.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <PreachingTable /> component', () => {
    (usePreaching as jest.Mock).mockReturnValue({
        state: preachingsStateMock,
        setSelectedPreaching: setSelectedPreachingMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors, selectedTheme: 'dark' },
    });

    beforeEach(() => {
        render(<PreachingTable />);
        jest.clearAllMocks();
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should table rows must match total preaching days', () => {

        /* Get rows of table */
        const rows = screen.getAllByTestId('preaching-table-row');
        expect(rows.length).toBe(preachingsStateMock.preachings.length);
    });

    it('should call setSelectedPreaching and navigate when row is pressed', () => {

        /* Get one row of table */
        const row = screen.getAllByTestId('preaching-table-row')[0];
        fireEvent.press(row);

        /**
         * Check if setSelectedPreaching and navigate is called one
         * time with respective args
         */
        expect(setSelectedPreachingMock).toHaveBeenCalledTimes(1);
        expect(setSelectedPreachingMock).toHaveBeenCalledWith(preachingsStateMock.preachings[0]);
        expect(navigateMock).toHaveBeenCalledTimes(1);
        expect(navigateMock).toHaveBeenCalledWith('AddOrEditPreachingScreen');
    });
});