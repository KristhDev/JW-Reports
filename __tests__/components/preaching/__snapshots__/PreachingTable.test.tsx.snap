// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <PreachingTable /> component should to match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#000000",
      "borderWidth": 1,
    }
  }
  testID="preaching-table"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#3C3547",
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
          },
          {
            "width": 121,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "marginLeft": 10,
          }
        }
        testID="table-cell-text"
      >
        N°
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#3C3547",
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
          },
          {
            "width": 121,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "marginLeft": 10,
          }
        }
        testID="table-cell-text"
      >
        Día
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#3C3547",
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
          },
          {
            "width": 242,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "marginLeft": 10,
          }
        }
        testID="table-cell-text"
      >
        Hora de Inicio
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#3C3547",
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
          },
          {
            "width": 242,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "marginLeft": 10,
          }
        }
        testID="table-cell-text"
      >
        Hora de Fin
      </Text>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          1
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          2
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          3
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          4
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          5
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          6
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          7
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          8
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          9
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          10
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="preaching-table-row"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          11
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 121,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          12
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          13:00
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#3C3547",
              "borderColor": "#000000",
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
            },
            {
              "backgroundColor": "#746C84",
              "width": 242,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          testID="table-cell-text"
        >
          16:00
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#3C3547",
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "#544C63",
            "width": 242,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "marginLeft": 10,
          }
        }
        testID="table-cell-text"
      >
        Total
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#3C3547",
            "borderColor": "#000000",
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "#544C63",
            "width": 484,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "marginLeft": 10,
          }
        }
        testID="table-cell-text"
      >
        33H
      </Text>
    </View>
  </View>
</View>
`;
